### DISCUSSION TITLE
How will Sidewalks and Doors Be Communicated?

### DISCUSSION THREAD

**Discussion by frastlin** [OvertureMaps/data#316] [Discussion]
Hello,
Looking on the [building](https://docs.overturemaps.org/reference/buildings/building) page, I see no information about doorways. On the [connector](https://docs.overturemaps.org/reference/transportation/connector) page, the stated purpose is to "create physical connections between Segments." This doesn't allow for a connector between a building and a segment to designate a door. I'm also not sure if places points should be used to designate doorways, as they seem to be mostly around identifying structures.

On the [segments](https://docs.overturemaps.org/reference/transportation/segment) documentation, there is no mention of how to connect a sidewalk segment to a road segment to identify what road the sidewalk borders. This could be OK if both the road segment and the sidewalk segment connected at a connector, but there would need to be clear rules about how only the parallel street should be connected to a sidewalk through a connector. This could also be a problem if the sidewalk only went partly along the road and started and ended in the middle of the road with bushes on either end.

**Comment by vcschapp** [comment]
Hi Brandon,
Thanks for raising these issues for discussion. I do some work on Overture's transportation schema task force, so I'm interested in continuing the transportation side of the discussion a bit further. (Hopefully one of our buildings gurus will pick up the buildings thread.)

The truth is we have a gap in the schema on the pedestrian side. We want to plug it in fairly short order, as pedestrian transportation is an important use case, but we still have some work to do.

If I'm understanding your question, it's about how to relate a sidewalk segment and a road segment when they are physically related in the real world but logically separated in the dataset. This makes a lot of sense.

I'm curious about a few things:

1. What is your use case, and why do you need to have this information? (Please provide as much detail as you feel comfortable sharing!)
2. Can you point to any existing models for this? (For example does OpenStreetMap have a conventional way of achieving this?)

**Comment by frastlin** [comment]
Thank you for your reply. Doorway and sidewalk information is for pedestrians, delivery drivers, rideshare, autonomous vehicles, city planning, and many other jobs that need to know about sidewalks and doorways. One example would be how I live in an apartment complex with 10 different buildings in it, and someone needs to deliver my food delivery order. The door is around 200 feet from the road along a sidewalk. There are many roads and parking lots in this complex, and even when I tell the drivers to meet me on the road, they often go the wrong direction. This is just one example that is related to me personally.

I'm creating a cross-sensory digital map viewer and it's important that the sidewalks have some kind of name that connects them to their street e.g., "Sidewalk to Fillmore St.". Otherwise, the name will be "Sidewalk", that is not very useful. When non-visual users move around the map, they rely on the contextual information they get from the feature name.

The [OpenSidewalks (OSW) project](https://github.com/OpenSidewalks/OpenSidewalks-Schema/tree/2023-doc-rerelease-and-rev) is under active development and is meant to support these kinds of entities. I know that each sidewalk and street has a starting ID and an ending ID (UID and VID), and there is a separate "node" that is a point that connects the sidewalk and the road segment. Nodes in OSW are like connectors. They also have points that could be doorways as well.

I know the [indoor mapping data format](https://www.mappedin.com/mapping/imdf-export/#:~:text=IMDF%20is%20a%20GeoJSON%20based,element%20of%20an%20indoor%20map.) has support for doorways.

I would think that a doorway would have a connector with a building, or even have the id of the building it is the doorway of.

There is an [ITU standard](https://www.itu.int/ITU-T/recommendations/rec.aspx?rec=13185&lang=en) that talks about how audio navigation applications should be developed, and this [Wayfindr Open Standard](https://www.wayfindr.net/open-standard) is very specific about the information needed related to doorways and sidewalks:

> #### 10.1 Entrances and exits 
> The audio instruction communications from an IABNNS should: 
> – utilize the most accessible entrance and exit; 
> – allow users to choose their entrance and exit; 
> – provide information about the type, size and opening configuration of doors; 
> – provide information about the position and type of door opening furniture and operating 
> controls. 
> 
> #### 10.2 Pathways 
> The audio instruction communications from an IABNNS should: 
> – utilize the most accessible pathway; 
> – allow users to choose their pathway; 
> – provide information about the type and size of the pathway; 
> – inform users of any curve in the pathway;
> – Where appropriate, advise users to keep on the side of the pathway that is best for the 
> direction in which they are travelling when on a route where people are moving in both 
> directions. (S 9.0.0.1)

**Comment by klayton-symbiosis** [comment]
Wheelchair and wheeled device accessibility data would also be nice (e.g. flags or restrictions indicating no ramp access, no sloping curb, etc.)
